### PR TITLE
docs: add Index Lifecycle Management page

### PR DESCRIPTION
**Pull request by wxyucs**
## Summary

Adds an **Index Lifecycle Management** page under Advanced Features that documents the `Remove`,
`UpdateVector` / `UpdateId`, `Clone`, and `ExportModel` operations. These APIs are exercised by
existing examples (`303_feature_remove.cpp`, `305_feature_update.cpp`, `309_feature_clone.cpp`,
`310_feature_export_model.cpp`) but had no counterpart on the website.

## What's covered

- Capability-flag matrix per index (HGraph / IVF / SINDI) for each operation.
- `RemoveMode::MARK_REMOVE` vs `RemoveMode::FORCE_REMOVE` semantics, plus the HGraph
  `support_remove` build parameter.
- `UpdateVector(..., force_update)` rejection semantics and the recommended `Remove` + `Add`
  fall-back pattern.
- `Clone` (with optional custom allocator) and `ExportModel` (training state without vectors)
  usage and runnable example pointers.

## Documentation set sync

- `docs/docs/en/src/advanced/index_lifecycle.md` (new)
- `docs/docs/zh/src/advanced/index_lifecycle.md` (new)
- `docs/docs/en/src/SUMMARY.md` (one navigation entry)
- `docs/docs/zh/src/SUMMARY.md` (one navigation entry)

No source or test changes; documentation-only PR.

### CODE REVIEW

File: docs/docs/en/src/advanced/index_lifecycle.md [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+### `UpdateVector`
+
+`UpdateVector(id, new_base, force_update = false)` replaces the vector data of an existing id in
+place. The default `force_update = false` mode performs a connectivity check: if the new vector
+is far from the original (which would degrade graph quality), the update is **rejected** and the
+caller is expected to fall back to `Remove` + `Add`.
+
+```cpp
+auto upd = vsag::Dataset::Make();
+upd->NumElements(1)->Dim(dim)->Ids(&id)->Float32Vectors(new_vec)->Owner(false);
```
[the remark this answers is not in the document]
Comment (wxyucs) [reply docs/docs/en/src/advanced/index_lifecycle.md]: Good catch — fixed. The English example now explicitly declares `std::vector<float> new_vec(dim);` and calls `Float32Vectors(new_vec.data())`, matching the style used in `examples/cpp/305_feature_update.cpp`.

File: docs/docs/en/src/advanced/index_lifecycle.md [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+A runnable example is available at `examples/cpp/309_feature_clone.cpp`.
+
+## Exporting the Trained Model
+
+`ExportModel()` returns an empty index that retains all trained state (quantization codebooks,
+centroids, hyperparameters) of the source but contains no vectors. It is the canonical way to
+share a pre-trained model across shards, processes, or hosts without re-running training.
+
+```cpp
+auto model = index->ExportModel().value();
```
[the remark this answers is not in the document]
Comment (wxyucs) [reply docs/docs/en/src/advanced/index_lifecycle.md]: Agreed. Replaced the `.value()` shortcut with an explicit `has_value()` check that returns early on failure, mirroring the `UpdateVector` example above.

File: docs/docs/zh/src/advanced/index_lifecycle.md [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+
+### `UpdateVector`
+
+`UpdateVector(id, new_base, force_update = false)` 在原地替换已有 id 对应的向量数据。默认的
+`force_update = false` 模式会做连通性检查：若新向量距离原向量较远（这会破坏图质量），更新会被
+**拒绝**，调用方应当退回到 `Remove` + `Add` 方案。
+
+```cpp
+auto upd = vsag::Dataset::Make();
+upd->NumElements(1)->Dim(dim)->Ids(&id)->Float32Vectors(new_vec)->Owner(false);
```
[the remark this answers is not in the document]
Comment (wxyucs) [reply docs/docs/zh/src/advanced/index_lifecycle.md]: 已采纳：中文示例同步声明 `std::vector<float> new_vec(dim);` 并使用 `Float32Vectors(new_vec.data())`，与 `examples/cpp/305_feature_update.cpp` 保持一致。

File: docs/docs/en/src/advanced/index_lifecycle.md [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+| `Clone`           | `SUPPORT_CLONE`                          |   Yes  | Yes |   —   |
+| `ExportModel`     | `SUPPORT_EXPORT_MODEL`                   |   Yes  | Yes |   —   |
+
+Always check at runtime with `index->CheckFeature(vsag::SUPPORT_*)` before calling these APIs;
+unsupported indexes return `UNSUPPORTED_INDEX_OPERATION`.
+
+## Removing Vectors
+
+`Remove` deletes vectors by id. HGraph must be built with `support_remove: true` in the
+`index_param` block; otherwise the call is rejected.
```
[the remark this answers is not in the document]
Comment (wxyucs) [reply docs/docs/en/src/advanced/index_lifecycle.md]: Verified — `HGraph::InitFeatures()` (src/algorithm/hgraph.cpp:1758) never sets `SUPPORT_DELETE_BY_ID`. Updated the capability matrix to drop that flag for `Remove` and replaced it with "_(no dedicated flag — see below)_". The surrounding text now directs readers to the "Removing Vectors" section to determine availability and which mode is supported, instead of relying on `CheckFeature`.

File: docs/docs/en/src/advanced/index_lifecycle.md [hunk 5]
```diff
@@ -0,0 +1,158 @@
...
+Each operation is optional and is exposed only when the underlying index advertises the matching
+capability flag via `index->CheckFeature(...)`.
+
+## Capability Flags
+
+| Operation         | Capability Flag                          | HGraph | IVF | SINDI |
+|-------------------|------------------------------------------|:------:|:---:|:-----:|
+| `Remove`          | `SUPPORT_DELETE_BY_ID`                   |   Yes  |  —  |   —   |
+| `UpdateVector`    | `SUPPORT_UPDATE_VECTOR_CONCURRENT`       |   Yes  |  —  |   —   |
+| `UpdateId`        | `SUPPORT_UPDATE_ID_CONCURRENT`           |   Yes  |  —  |   —   |
```
[the remark this answers is not in the document]
Comment (wxyucs) [reply docs/docs/en/src/advanced/index_lifecycle.md]: Good catch — `SINDI::InitFeatures()` (src/algorithm/sindi/sindi.cpp:731-732) does set both flags. Updated the matrix: SINDI is now marked "Yes" for both `UpdateVector` and `UpdateId`. (IVF still does not set either flag, so its column stays as "—".)

File: docs/docs/en/src/advanced/index_lifecycle.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (wxyucs) [reply docs/docs/en/src/advanced/index_lifecycle.md]: Confirmed against `HGraph::Remove` (src/algorithm/hgraph.cpp:1934): `MARK_REMOVE` only calls `LabelTable::MarkRemove` and works regardless of `support_remove`, while `FORCE_REMOVE` invokes `force_remove_one` → `graph_force_remove_one` which depends on the delete-tracking metadata gated by `support_remove: true`. Rewrote the "Removing Vectors" section to split the two modes explicitly, added a per-mode requirement column in the `RemoveMode` table, and noted that the `support_remove: true` JSON snippet is only required for `FORCE_REMOVE`.

File: docs/docs/zh/src/advanced/index_lifecycle.md [hunk 6]
```diff
@@ -0,0 +1,148 @@
...
+| `Clone`           | `SUPPORT_CLONE`                          |   是   | 是  |   —   |
+| `ExportModel`     | `SUPPORT_EXPORT_MODEL`                   |   是   | 是  |   —   |
+
+在调用上述 API 之前，请始终通过 `index->CheckFeature(vsag::SUPPORT_*)` 在运行时进行检查；
+不支持的索引会返回 `UNSUPPORTED_INDEX_OPERATION`。
+
+## 删除向量
+
+`Remove` 按 id 删除向量。HGraph 必须在 `index_param` 块中以 `support_remove: true` 构建，
+否则调用会被拒绝。
```
[the remark this answers is not in the document]
Comment (wxyucs) [reply docs/docs/zh/src/advanced/index_lifecycle.md]: 已核对 `HGraph::InitFeatures()`（src/algorithm/hgraph.cpp:1758），确实未声明 `SUPPORT_DELETE_BY_ID`。中文版能力矩阵已同步：`Remove` 行的能力标志改为「（暂无专用标志，参见下文）」，并在能力标志段落后补充说明 `Remove` 是否可用与支持哪种模式请看「删除向量」一节，不再依赖 `CheckFeature`。

File: docs/docs/zh/src/advanced/index_lifecycle.md [hunk 7]
```diff
@@ -0,0 +1,148 @@
...
+
+每个操作均为可选项，仅当底层索引通过 `index->CheckFeature(...)` 公布对应能力标志时才可用。
+
+## 能力标志
+
+| 操作              | 能力标志                                  | HGraph | IVF | SINDI |
+|-------------------|------------------------------------------|:------:|:---:|:-----:|
+| `Remove`          | `SUPPORT_DELETE_BY_ID`                   |   是   |  —  |   —   |
+| `UpdateVector`    | `SUPPORT_UPDATE_VECTOR_CONCURRENT`       |   是   |  —  |   —   |
+| `UpdateId`        | `SUPPORT_UPDATE_ID_CONCURRENT`           |   是   |  —  |   —   |
```
[the remark this answers is not in the document]
Comment (wxyucs) [reply docs/docs/zh/src/advanced/index_lifecycle.md]: 已修正：中文版能力矩阵将 SINDI 的 `UpdateVector` 与 `UpdateId` 标为「是」，对应 `SINDI::InitFeatures()`（src/algorithm/sindi/sindi.cpp:731-732）开启的两个 flag。IVF 仍未声明这两个 flag，因此该列保持「—」。

File: docs/docs/zh/src/advanced/index_lifecycle.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (wxyucs) [reply docs/docs/zh/src/advanced/index_lifecycle.md]: 已对照 `HGraph::Remove`（src/algorithm/hgraph.cpp:1934）核实：`MARK_REMOVE` 仅调用 `LabelTable::MarkRemove`，不依赖 `support_remove`；`FORCE_REMOVE` 调用 `force_remove_one` → `graph_force_remove_one`，依赖由 `support_remove: true` 配置的删除追踪元数据。中文版「删除向量」一节已改写：分两条列出两种模式的不同要求，`RemoveMode` 表格中补上每种模式的要求列，并说明示例 JSON 中的 `support_remove: true` 仅在使用 `FORCE_REMOVE` 时是必需的。

**Review by LHT129 — APPROVED** [review]
LGTM
